### DISCUSSION TITLE
TCl().eval('info library') gives previous build's install location

### DISCUSSION THREAD

**Discussion by elibroftw** [pyinstaller/pyinstaller#6278] [Help] [answered]
Hey. I use pyinstaller to bundle my app and I combine this with Inno Setup.
The app installs to "C:\Users\maste\AppData\Local\Programs\Music Caster".
Now when I'm building, tcl cannot be found and I narrowed down the issue to:
```py
from tkinter import Tcl
print(Tcl().eval("info library"))
```
The output is `C:\Users\maste\AppData\Local\Programs\Music Caster\tcl` which is not the expected `C:\Users\maste\AppData\Local\Programs\Python\Python38-32\tcl\tcl8.6`. 

I fixed the issue my repairing the interpreter but what could've caused this to begin with?

**Comment by bwoodsend** [comment]
Why isn't `C:\Users\maste\AppData\Local\Programs\Music Caster\tcl` where you expect tcl to be? PyInstaller should have collected tcl at build time and added it to your app and there it is. Why would you expect it to use the original copy of tcl?

**Reply by elibroftw** [reply]
Well I expect tcl to be there, but I don't expect Python to use that tcl instead of the original copy. PyInstaller did not collect tcl with the unexpected location.

When I built on my laptop, Python uses the original copy. When I built on my Desktop before repairing Python, my app's tcl was used by Python.

Could adding the the app's install location to %PATH% have caused this problem? I had removed it from %PATH% but the issue persisted.

**Reply by bwoodsend** [reply]
You mean that your problem is running `print(Tcl().eval("info library"))` on normal unfrozen Python has started to use `C:\Users\maste\AppData\Local\Programs\Music Caster\tcl`? In which case, yes, adding an app's install location to `%PATH%` leaves Windows in a bit of a free for all state - whichever folder appears first dictates which copy of tcl will get used.

**Reply by rokm** [reply]
Do you happen to have `TCL_LIBRARY` or `TK_LIBRARY` environment variable set for some reason? PyInstaller-frozen programs using `tcl` set those at runtime, but they should not be propagating outside of their own processes...

> Could adding the the app's install location to %PATH% have caused this problem? I had removed it from %PATH% but the issue persisted.

Adding to `PATH` might cause python to use the tcl/tk DLLs from the app bundle instead of its own, which could affect the Tcl/Tk library (as in, tcl scripts library, not DLL shared library) search paths.

**Reply by elibroftw** [reply]
No TCL_LIBRARY and TK_LIBRARY are not set. If it is the %PATH%, then my only question is how come removing my app's install path from %PATH% did not fix the issue even after restarting VS Code to rerun the snippet? Do you think if I wanted to launch my app from command line I should instead create a folder with nothing but a single program/script?

**Reply by rokm** [reply]
> If it is the %PATH%, then my only question is how come removing my app's install path from %PATH% did not fix the issue even after restarting VS Code to rerun the script? 

How did you modify the `PATH`? And did you check if the restarted instance of VS Code was indeed using the new PATH (e.g., by printing out `os.environ['PATH']` in python)?

**Reply by elibroftw** [reply]
I had modified path manually through environment variable settings.

I didn't check if the updated path was used. It's not a big deal anymore. Now I know to not add the install directory to path.

**Comment by elibroftw** [comment, marked as answer]
Not an issue anymore. Thanks for all the suggestions. I just repaired the Python intepretor after removing my apps install Dir from %PATH%.
